### PR TITLE
perf: Optimize `CssSizeLength` & `CssStyleBuilder`

### PR DESCRIPTION
**Pull request by LeaFrock**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [x] Performance optimization
- [ ] Refactoring
- [x] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Optimize `CssSizeLength` & `CssStyleBuilder` |
| 🇨🇳 Chinese | 优化`CssSizeLength` 和`CssStyleBuilder` |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/core/CssSizeLength.cs [hunk 1]
```diff
@@ -145,5 +139,66 @@ public static bool TryParse(string value, out CssSizeLength cssSizeLength)
...
+            {
+                unit = CssSizeLengthUnit.NoUnit;
+                return true;
+            }
+
+            if (!IsNumberOrDot(trimmedValue[0]))
+            {
+                unit = CssSizeLengthUnit.Calc;
+                return true;
+            }
```
Comment (ElderJames) [inline components/core/CssSizeLength.cs:155]: Is the `unit` result backwards?

Comment (LeaFrock) [reply components/core/CssSizeLength.cs:155]: What does `backwards` mean? The logic should be the same as before.

```csharp
            if (value.StartsWith("calc", StringComparison.OrdinalIgnoreCase))
            {
                _stringValue = value;
                _value = null;
                _unit = CssSizeLengthUnit.NoUnit;
                return;
            }

            var index = value
                .Select((c, i) => ((char c, int i)?)(c, i))
                .FirstOrDefault(x => !(x.Value.c == '.' || x.Value.c >= '0' && x.Value.c <= '9'))
                ?.i
                ?? value.Length;

            if (index == 0) // The first char is not a number or dot. Or the value is an empty string.
            {
                _stringValue = value;
                _value = null;
                _unit = CssSizeLengthUnit.Calc;
                _parsed = false;
                return;
            }
```

Comment (ElderJames) [reply components/core/CssSizeLength.cs:155]: I see, but we may be wrong before. If `trimmedValue` starts with `calc`, the unit should be `Calc`, and shoud be `NoUnit` while it not starts with number or dot.

Comment (LeaFrock) [reply components/core/CssSizeLength.cs:155]: > but we may be wrong before

🤣 Really? Please make sure about it and if so, I'll make a commit for this. Also, we need to check if it's a lack of related unit tests which did not expose the issue before.

Comment (LeaFrock) [reply components/core/CssSizeLength.cs:155]: The changes are related to [your commit](https://github.com/ant-design-blazor/ant-design-blazor/commit/c8b52156a051bb10e17c4719262bf6524cb03913#diff-0a92621cbff66fd618c8848c2cd86ad991bb9a4b20cddaab1b9ca0fabfa2ee06) @ElderJames  .

Comment (ElderJames) [reply components/core/CssSizeLength.cs:155]: Logically, that's true. 😅

Comment (LeaFrock) [reply components/core/CssSizeLength.cs]: Fixed.

**Review by ElderJames — APPROVED** [review]

**Pull request merged**